src/kbmod/search/ImageStack.cpp

```diff
@@ -62,21 +62,13 @@ void ImageStack::setTimeOrigin() {
     for (auto& t : imageTimes) t = t - initialTime;
 }
 
-std::vector<LayeredImage>& ImageStack::getImages() { return images; }
-
-unsigned ImageStack::imgCount() const { return images.size(); }
-
-const std::vector<float>& ImageStack::getTimes() const { return imageTimes; }
-
-float* ImageStack::getTimesDataRef() { return imageTimes.data(); }
-
 LayeredImage& ImageStack::getSingleImage(int index) {
-    if (index < 0 || index > images.size()) throw std::runtime_error("ImageStack index out of bounds.");
+    if (index < 0 || index > images.size()) throw std::out_of_range("ImageStack index out of bounds.");
```

Comment:
Is there something useful to add to the message? I like to format these so that the grep text is early "ImageStack index out of bounds" followed by exception specific information useful to the user. In this case it might be the actual index and allowed ranges and if available the affected object. The idea is to make diagnostics more helpful. Could also be done in the python layer of course.